maintenance_security/models/mail_followers.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import fields, models
+
+
+class Followers(models.Model):
+    _inherit = "mail.followers"
+
+    def _add_followers(
+        self,
+        res_model,
+        res_ids,
+        partner_ids,
+        subtypes,
+        check_existing=False,
+        existing_policy="skip",
+    ):
+        """We need to prevent a partner without Maintenance permission from being
```

Comment:
This is preventing to be added as follower, but not the special case that is embedded in `mail` module, that when you set the field `user_id`, a mail with subject "You have been assigned to" is sent. Please restrict also this one.

Reply:
Sorry, probably have to explain more the reason for this. `maintenance.equipment` does not have the `user_id` field, similar field is https://github.com/odoo/odoo/blob/15.0/addons/maintenance/models/maintenance.py#L117 and that field has `tracking=true`. 

With each change in that field https://github.com/odoo/odoo/blob/15.0/addons/maintenance/models/maintenance.py#L212 becomes a follower of a specific subtype https://github.com/odoo/odoo/blob/15.0/addons/maintenance/models/maintenance.py#L90

The email that is sent is for the `mail.tracking` record of this change, and for this reason the solution has had to be this, to prevent it from being added as a follower of that subtype.